torchmetrics/metric.py

```diff
@@ -180,6 +180,10 @@ def forward(self, *args, **kwargs):
 
     def _sync_dist(self, dist_sync_fn=gather_all_tensors):
         input_dict = {attr: getattr(self, attr) for attr in self._reductions.keys()}
+        for attr, reduction_fn in self._reductions.items():
+            # dim_zero_cat can be applied multiple times, this optimizes the number of gathers
+            if reduction_fn == dim_zero_cat and len(input_dict[attr]) > 1:
```

Comment:
in principal there is nothing preventing the the user from having metric state as a tensor and also set `dist_sync_fx='cat'`. To be sure we do not mess things up we probably have to also add condition that it should be a list:
```suggestion
            if reduction_fn == dim_zero_cat and isinstance(input_dict[attr]) and len(input_dict[attr]) > 1:
```